brave-spring-resttemplate-interceptors/src/test/java/com/github/krisofa/brave/client/BraveClientHttpRequestInterceptorTest.java

```diff
@@ -0,0 +1,128 @@
+package com.github.krisofa.brave.client;
+
+import com.github.kristofa.brave.ClientRequestInterceptor;
+import com.github.kristofa.brave.ClientResponseInterceptor;
+import com.github.kristofa.brave.ClientTracer;
+import com.github.kristofa.brave.SpanId;
+import com.github.kristofa.brave.http.SpanNameProvider;
+import com.github.kristofa.brave.http.StringServiceNameProvider;
+import org.junit.Test;
+import org.mockito.InOrder;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.client.ClientHttpRequestExecution;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.mock.http.client.MockClientHttpRequest;
+import org.springframework.mock.http.client.MockClientHttpResponse;
+
+import java.io.IOException;
+import java.net.URI;
+
+import static org.apache.commons.lang3.RandomStringUtils.randomAlphanumeric;
+import static org.junit.Assert.assertSame;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.inOrder;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class BraveClientHttpRequestInterceptorTest {
+
+    private final ClientTracer clientTracer = mock(ClientTracer.class);
+    private final String serviceName = randomAlphanumeric(20);
+    private final SpanNameProvider spanNameProvider = mock(SpanNameProvider.class);
+    private final BraveClientHttpRequestInterceptor subject = new BraveClientHttpRequestInterceptor(new ClientRequestInterceptor(clientTracer),
```

Comment:
@wjam I think you can improve the test by mocking `ClientRequestInterceptor` and `ClientResponseInterceptor`. Now you are going one layer deeper by mocking `ClientTracer` which means you are also testing / verifying the interceptor which seems unneeded.